### PR TITLE
Investigator check/installs host requirements.

### PR DESCRIPTION
**Pull request by ashcrow**
- Installs docker if not present
- Updates docker configuration with templated config
- Installs kubernetes if not present
- Updates flanneld configuration with templated config
- Installs flannel if not present
- Updates kubelet and kube_config configuration with templated config
- Updates ansible dep to 2.0.0.2

### CODE REVIEW

File: src/commissaire/jobs/investigator.py [hunk 1]
```diff
@@ -19,10 +19,33 @@
...
+
+
+def clean_up_key(key_file):
+    """
+    Remove the key file.
+
+    :param key_file: Full path to the key_file
+    :type key_file: str
+    :param logger: The logger
+    :type logger: logging.Logger
```
Comment (mbarnes) [inline src/commissaire/jobs/investigator.py]: Looks like this can be deleted?  (no `logger` param)

File: src/commissaire/oscmd/atomic.py [hunk 2]
```diff
@@ -44,3 +44,39 @@ def upgrade(self):
...
+        Atomic install docker command.
+
+        :return: The command to execute as a list
+        :rtype: list
+        """
+        return []
+
+    def start_docker(self):
+        """
+        Atomic start start docker command..
```
Comment (mbarnes) [inline src/commissaire/oscmd/atomic.py]: Couple typos here?

File: src/commissaire/oscmd/fedora.py [hunk 3]
```diff
@@ -44,3 +44,39 @@ def upgrade(self):
...
+        Fedora install docker command.
+
+        :return: The command to execute as a list
+        :rtype: list
+        """
+        return ['dnf', 'install', '-y', 'docker']
+
+    def start_docker(self):
+        """
+        Fedora start start docker command..
```
Comment (mbarnes) [inline src/commissaire/oscmd/fedora.py]: Couple typos here?

File: src/commissaire/oscmd/redhat.py [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+from commissaire.oscmd.rhel import OSCmd as OSCmdBase
+
+
+class OSCmd(OSCmdBase):
+    """
+    Commmands for RHEL.
+    """
+
+    #: The type of Operating System
+    os_type = 'redhat'
```
Comment (mbarnes) [inline src/commissaire/oscmd/redhat.py]: Is this class just meant to be an alias for `rhel`?

File: src/commissaire/transport/ansibleapi.py [hunk 5]
```diff
@@ -166,7 +172,16 @@ def _run(self, ip, key_file, play_source):
         finally:
             if tqm is not None:
                 tqm.cleanup()
-        return result
+
+        if result in expected_results:
+            fact_cache = {}
+            self.logger.debug('{0}: Good result {1}'.format(ip, result))
+            if ip in self.variable_manager._fact_cache.keys():
+                fact_cache = self.variable_manager._fact_cache[ip]
```
Comment (mbarnes) [inline src/commissaire/transport/ansibleapi.py]: Minor, but could simplify this slightly:
`fact_cache = self.variable_manager._fact_cache.get(ip, {})`

File: requirements.txt [hunk 6]
```diff
@@ -1,5 +1,7 @@
 gevent==1.1rc3
 falcon>=0.3
+requests
 bcrypt
 python-etcd==0.4.2
-ansible>=2.0.0.1
\ No newline at end of file
+setuptools
+ansible>=2.0.0.2
```
Comment (mbarnes) [inline requirements.txt]: Missing `jinja2`?

File: doc/endpoints.rst [hunk 7]
```diff
@@ -237,6 +237,7 @@ Creates a new host record.
    {
        "address": string      // The IP address of the cluster host
        "ssh_priv_key": string // base64 encoded ssh private key
+       "cluster": string      // The cluster the host should be associated with
```
Comment (mbarnes) [inline doc/endpoints.rst]: This looks like it slipped into the branch.  Noted for myself, but unrelated to `bootstrap`?

File: src/commissaire/oscmd/redhat.py (hunk 4 above)
Comment (ashcrow) [inline src/commissaire/oscmd/redhat.py]: Correct.

File: requirements.txt (hunk 6 above)
Comment (ashcrow) [inline requirements.txt]: It gets pulled as a dep but you are right, it should be in there. I'll add it.

**Comment by mbarnes** [comment]
Gotta defer to @cooktheryan on the Ansible stuff, but LGTM aside from some nits.

File: requirements.txt (hunk 6 above)
Comment (ashcrow) [inline requirements.txt]: Added.

File: doc/endpoints.rst (hunk 7 above)
Comment (ashcrow) [inline doc/endpoints.rst]: Yeah it's unrelated to bootstrap. I'll yank it off this branch if you don't mind throwing it in your PR.

**Comment by mbarnes** [comment]
:+1: to the fixups.

Captured the cluster docs bit to a new branch where I'll implement it.

**Comment by ashcrow** [comment]
Squashed fixups.

**Pull request merged**